internal/process/provisioning/edp_registration.go

```diff
@@ -89,30 +94,29 @@ func (s *EDPRegistrationStep) Run(operation internal.Operation, log *slog.Logger
 		op.EDPCreated = true
 	}, log)
 	if repeat != 0 {
-		log.Error("cannot save operation")
-		return newOp, 5 * time.Second, nil
+		log.Error("cannot update operation")
+		return s.operationManager.RetryOperation(newOp, "cannot update operation", err, dbRetryInterval, dbRetryTimeout, log)
 	}
 
 	return newOp, 0, nil
 }
 
 func (s *EDPRegistrationStep) handleError(operation internal.Operation, err error, log *slog.Logger, msg string) (internal.Operation, time.Duration, error) {
 	log.Warn(fmt.Sprintf("%s: %s", msg, err))
-
-	if kebError.IsTemporaryError(err) {
-		since := time.Since(operation.UpdatedAt)
-		if since < time.Minute*30 {
+	if s.config.Required {
```

Comment:
Can we check temporary error first to avoid duplication in lines 107 and 113 as it is done in the original code?

Reply:
We can, but I decided that this is easier to understand despite duplication.
We have clearly separated paths for different configs.
Either way the duplication remains. 
This needs to be addressed in timeouts refactoring story.

Reply:
```go
	if kebError.IsTemporaryError(err) {
                if s.config.Required {
                   RetryOperationWithoutFail()
                } else {
                   RetryOperation()
		}
	}

	if s.config.Required {
        	return s.operationManager.OperationFailed(operation, msg, err, log)
	} else {
		log.Warn(fmt.Sprintf("Step %s failed. Step is not required. Skip step.", s.Name()))
		return operation, 0, nil
        }
```